File: frontend/src/core/services/resena.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { environment } from '../../environments/environment';
import { ApiResponse } from './sucursal.service';

export interface ResenaRequest {
  solicitudEnvioId: number;
  usuarioId: number;
  calificacion: number;
  comentario: string;
}

export interface Resena {
  id: number;
  codigoSeguimiento: string;
  usuarioNombre: string;
  calificacion: number;
  comentario: string;
  fechaCreacion: string;
}

@Injectable({
  providedIn: 'root'
})
export class ResenaService {

  private apiUrl = `${environment.apiUrl}/resenas`;

  constructor(private http: HttpClient) {}

  listar(): Observable<ApiResponse<Resena[]>> {
    return this.http.get<ApiResponse<Resena[]>>(this.apiUrl);
  }

  crear(request: ResenaRequest): Observable<ApiResponse<Resena>> {
    return this.http.post<ApiResponse<Resena>>(this.apiUrl, request);
  }
}

File: ramona-microservices/resena-service/src/main/java/cl/ramona/resenaservice/service/ResenaService.java
package cl.ramona.resenaservice.service;

import cl.ramona.resenaservice.dto.request.ResenaRequest;
import cl.ramona.resenaservice.dto.response.ResenaResponse;
import java.math.BigDecimal;
import java.util.List;

public interface ResenaService {
    ResenaResponse crear(ResenaRequest request);
    List<ResenaResponse> listar();
    List<ResenaResponse> listarPorUsuario(Long usuarioId);
    ResenaResponse obtenerPorSolicitud(Long solicitudEnvioId);
    BigDecimal promedioCalificacion();
}
